### DISCUSSION TITLE
PKCE Login Flow without 3rd Party Provider?

### DISCUSSION THREAD

**Discussion by Dark-Akkula** [supabase/supabase#20357] [Questions]
I am pretty confused by the documentation on this.  I understand that this is a more secure version of Authentication flow where you login with your ID and password and instead of returning back the access token for Supabase directly, it send a code back.  The code then must be sent back to a URL set up to handle it and that causes the access token to be issued.  This is more secure than the implicit flow giving the access token directly.

The question I have with the documents is that I understand all of this is simply handled based on responses to a designated URL with the authorization code and access token.  This is similar to what I already have set up with Amazon cognito.  However the Supabase docs talk about storing cookies and all kinds of stuff like that.  I don't understand why that is even required or why these codes aren't just returned in the URL upon redirect.  

I also don't understand the .signInWithOAuth function because it request a provider like google or one of the third party providers.  Is that a requirement to use this or can we simply skip using the provider and just use internal Supabase functions.  

Is there an example of how to set this up without using all the random third party identity providers?  Thanks!

**Comment by Dark-Akkula** [comment]
Does anybody know if the PKCE flow is even possible without a 3rd party auth provider?

**Reply by J0** [reply]
Hey, thanks for the query!

>I am pretty confused by the documentation on this. I understand that this is a more secure version of Authentication flow where you login with your ID and password and instead of returning back the access token for Supabase directly, it send a code back. The code then must be sent back to a URL set up to handle it and that causes the access token to be issued. This is more secure than the implicit flow giving the access token directly.

Sounds accurate. 

>However the Supabase docs talk about storing cookies and all kinds of stuff like that. I don't understand why that is even required or why these codes aren't just returned in the URL upon redirect.

We currently use the PKCE flow which has a code verifier which is stored in the Cookies. Not too familiar with Cognito but my guess is that AWS is using the Auth Code flow, which doesn't require a cookie

> I also don't understand the .signInWithOAuth function because it request a provider like google or one of the third party providers. Is that a requirement to use this or can we simply skip using the provider and just use internal Supabase functions.

You'll need a third party provider for sign in with OAuth as the third party provider (e.g. Google) provides your identity on that platform to Supabase as a way of authenticating you.

**Reply by J0** [reply]
You can use PKCE Flow with any of the regular flows (e.g. magic link, OTP)
